src/utils/risk.ts

```diff
@@ -0,0 +1,275 @@
+import {
+    compareNumber,
+    isDefined,
+    isFalsyString,
+    isNotDefined,
+    listToGroupList,
+    mapToList,
+    mapToMap,
+    unique,
+} from '@togglecorp/fujs';
+import { sumSafe, maxSafe, avgSafe } from '#utils/common';
+import type { paths, components } from '#generated/riskTypes';
+import {
+    CATEGORY_RISK_HIGH,
+    CATEGORY_RISK_VERY_HIGH,
+    CATEGORY_RISK_MEDIUM,
+    CATEGORY_RISK_LOW,
+    CATEGORY_RISK_VERY_LOW,
+    COLOR_HAZARD_CYCLONE,
+    COLOR_HAZARD_DROUGHT,
+    COLOR_HAZARD_EARTHQUAKE,
+    COLOR_HAZARD_FLOOD,
+    COLOR_HAZARD_FOOD_INSECURITY,
+    COLOR_HAZARD_STORM,
+    COLOR_HAZARD_WILDFIRE,
+    COLOR_LIGHT_GREY,
+} from './constants';
+
+type HazardType = components['schemas']['HazardTypeEnum'];
+
+export const hazardTypeToColorMap: Record<HazardType, string> = {
+    EQ: COLOR_HAZARD_EARTHQUAKE,
+    FL: COLOR_HAZARD_FLOOD,
+    TC: COLOR_HAZARD_CYCLONE,
+    EP: COLOR_LIGHT_GREY,
+    FI: COLOR_HAZARD_FOOD_INSECURITY,
+    SS: COLOR_HAZARD_STORM,
+    DR: COLOR_HAZARD_DROUGHT,
+    TS: COLOR_HAZARD_CYCLONE,
+    CD: COLOR_LIGHT_GREY,
+    WF: COLOR_HAZARD_WILDFIRE,
+};
+
+export function getDataWithTruthyHazardType<
+    HAZARD_TYPE extends HazardType | '',
+    DATA extends { hazard_type?: HAZARD_TYPE | undefined | null }
+>(data: DATA) {
+    if (isFalsyString(data.hazard_type)) {
+        return undefined;
+    }
+
+    return {
+        ...data,
+        hazard_type: data.hazard_type as HazardType,
+    };
+}
+
+export interface RiskDataItem {
+    january?: number | null,
+    february?: number | null,
+    march?: number | null,
+    april?: number | null,
+    may?: number | null,
+    june?: number | null,
+    july?: number | null,
+    august?: number | null,
+    september?: number | null,
+    october?: number | null,
+    november?: number | null,
+    december?: number | null,
+    annual_average?: number | null,
+}
+
+export const monthNumberToNameMap: Record<number, keyof RiskDataItem> = {
+    0: 'january',
+    1: 'february',
+    2: 'march',
+    3: 'april',
+    4: 'may',
+    5: 'june',
+    6: 'july',
+    7: 'august',
+    8: 'september',
+    9: 'october',
+    10: 'november',
+    11: 'december',
+    12: 'annual_average',
```

Comment:
Can we not avoid this hack?

Reply:
What's the hack here?

Reply:
Using 12 for something that is not a month.